### PR TITLE
Add MetaproSIP

### PR DESCRIPTION
**Pull request by bernt-matthias**
Hope that this will be the first of many OpenMS workflows @ IWC (and also my first contribution).

Screenshot from 2023-01-31 13-31-02

- [x] add tests
  - https://abibuilder.cs.uni-tuebingen.de/archive/openms/galaxy-testdata/metaprosip/
  - normal 13C SIP with standard parameter
- [x] Will the category `proteomics` fit for OpenMS workflows or should it be something like `mass_spectrometry`?
- [ ] Are any intermediate outputs of interest?
- [x] Are there any other workflow parameters?
- [x] Tests
- [x] .dockstore.yml
- [x] CHANGELOG
- [x] Change name to just `MetaProSIP`
- [ ] ...

### CODE REVIEW

**Comment by mvdbeek** [comment]
let me know if there's anything needed from our side to push this further

**Comment by bernt-matthias** [comment]
Asked the OpenMS devs for input regarding README, desired outputs and needed inputs...

File: workflows/proteomics/openms-metaprosip/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+- Fasta (aminoacid sequences)
+
+## Inputs values
+
+## Processing
+
+- DecoyDatabase: Add decoy sequences to the Fasta database (for FDR calculation)
+- FeatureFinderCentroided: identify eluting peptides that correspond to isotopologues with natural isotopic distributions 
+- MSGFPlusAdapter: identify peptides through peptide fragment fingerprinting (database search)
+- FeatureFinderMultiplex
```
Comment (timosachsenberg) [inline workflows/proteomics/openms-metaprosip/README.md:15]:
```suggestion
- FeatureFinderMultiplex: detect elution profiles of unlabeled peptides
```

File: workflows/proteomics/openms-metaprosip/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+- PeptideIndexer
```
Comment (timosachsenberg) [inline workflows/proteomics/openms-metaprosip/README.md:16]:
```suggestion
- PeptideIndexer: annotate protein association to identified peptides
```

File: workflows/proteomics/openms-metaprosip/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+- FalseDiscoveryRate: Calculate FDR
+- IDMapper
```
Comment (timosachsenberg) [inline workflows/proteomics/openms-metaprosip/README.md:18]:
```suggestion
- IDMapper: map identified spectra to elution profiles
```

File: workflows/proteomics/openms-metaprosip/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# MetaProSIP: automated inference of elemental fluxes in microbial communities
+
+## Inputs dataset
+
+- mzML
+- Fasta (aminoacid sequences)
+
+## Inputs values
+
```
[the remark this answers is not in the document]
Comment (bernt-matthias) [reply workflows/proteomics/openms-metaprosip/README.md:9]: Thanks for the suggestion. I will add those as workflow parameters.

[the remark this answers is not in the document]
Comment (bernt-matthias) [reply workflows/proteomics/openms-metaprosip/README.md:9]: Does MetaProSIP's parameter `-mz_tolerance_ppm` should be the same as MSGF+'s `-precursor_mass_tolerance`?

[the remark this answers is not in the document]
Comment (timosachsenberg) [reply workflows/proteomics/openms-metaprosip/README.md:9]: I think this is a sensible choice.

**Comment by bernt-matthias** [comment]
@timosachsenberg seems that Galaxy workflows do not allow (yet) for workflow inputs connecting to selects allowing to select multiple options. 

Here we seem to need this for the fixed and variable modifications. Just to be sure: selecting multiple modifications is relevant, or?

**Comment by timosachsenberg** [comment]
> @timosachsenberg seems that Galaxy workflows do not allow (yet) for workflow inputs connecting to selects allowing to select multiple options.
> 
> Here we seem to need this for the fixed and variable modifications. Just to be sure: selecting multiple modifications is relevant, or?

Unfortunately, yes

**Comment by mvdbeek** [comment]
I'll take a look at implementing multiple options, that seems kind of important.

**Comment by bernt-matthias** [comment]
Seems that the Galaxy release (23.1 which should be released soon) has the needed support for selects. So this is ready for review.

This is now for OpenMS 2.8. Guess I will need a bit until the OpenMS 3.0 galaxy tools will be ready -- so I would suggest to proceed anyway.

File: workflows/proteomics/openms-metaprosip/metaprosip.ga [hunk 5]
```diff
@@ -0,0 +1,624 @@
...
+    "annotation": "Automated inference of stable isotope incorporation rates in proteins for functional metaproteomics ",
+    "creator": [
+        {
+            "class": "Person",
+            "identifier": "0000-0003-3763-0797",
+            "name": "Matthias Bernt"
+        }
+    ],
+    "format-version": "0.1",
+    "license": "MIT",
```
Comment (mvdbeek) [inline workflows/proteomics/openms-metaprosip/metaprosip.ga:12]:
```suggestion
    "format-version": "0.1",
    "release": "0.1",
    "license": "MIT",
```

File: workflows/proteomics/openms-metaprosip/.dockstore.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+version: 0.1
```
Comment (mvdbeek) [inline workflows/proteomics/openms-metaprosip/.dockstore.yml:1]: This is the version of the dockstore yml file
```suggestion
version: 1.2
```

File: workflows/proteomics/openms-metaprosip/metaprosip-tests.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      filetype: fasta
+    Centroided LC-MS datasets:
+      class: Collection
+      collection_type: list
+      elements:
+      - class: File
+        identifier: Zeitz_SIP_13-II_020_picked.mzML
+        location: https://abibuilder.cs.uni-tuebingen.de/archive/openms/galaxy-testdata/metaprosip/Zeitz_SIP_13-II_020_picked.mzML
+        filetype: mzml
+    Precursor monoisotopic mass tolerance (ppm): '10.0'
```
Comment (mvdbeek) [inline workflows/proteomics/openms-metaprosip/metaprosip-tests.yml:15]:
```suggestion
    Precursor monoisotopic mass tolerance (ppm): 10.0
```

File: workflows/proteomics/openms-metaprosip/metaprosip-tests.yml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    Feature fitting result:
+      element_tests:
+        Zeitz_SIP_13-II_020_picked.mzML:
+          asserts:
+          - that: "has_n_lines"
+            n: 4091
+          - that: "has_text"
+            text: "Group 1"
+          - that: "has_text"
+            text: "LYSFHTLHQTYMK"
```
Comment (mvdbeek) [inline workflows/proteomics/openms-metaprosip/metaprosip-tests.yml:39]:
```suggestion
          asserts:
          - that: has_n_lines
            n: 85
          - that: has_text
            text: Peptide Sequence
          - that: has_text
            text: AATGTPSPAGSPPPIVPAPK
    Feature fitting result:
      element_tests:
        Zeitz_SIP_13-II_020_picked.mzML:
          asserts:
          - that: has_n_lines
            n: 4091
          - that: has_text
            text: Group 1
          - that: has_text
            text: LYSFHTLHQTYMK 
```
pretty minor, doesn't have any impact and deserializes to the same dict

File: workflows/proteomics/openms-metaprosip/.dockstore.yml [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+workflows:
+- name: main
+  primaryDescriptorPath: /metaprosip.ga
+  subclass: Galaxy
+  testParameterFiles:
+  - /metaprosip-tests.yml
```
Comment (mvdbeek) [inline workflows/proteomics/openms-metaprosip/.dockstore.yml:7]:
```suggestion
  - /metaprosip-tests.yml
  authors:
  - name: Matthias Bernt
    orcid: 0000-0003-3763-0797
```

(this should be autogenerated by planemo too, hopefully)

**Review by mvdbeek — COMMENTED** [review]
Super cool, I haven't checked this in the Galaxy interface yet but looks great so far.

**Comment by mvdbeek** [comment]
Alright, I had a quick look at this and it'd be nice to address the best practice panel recommendations:

Screenshot 2023-09-15 at 13 06 17

Note that every dataset seems to be a workflow output in the workflow, I'm not sure if this is intended or not. If it is intended it would be good to give this a descriptive label, if these are not the primary outputs of the workflow you can remove the checkmark:
Screenshot 2023-09-15 at 13 08 48

I assume the labelled outputs ("Feature fitting result" and "Peptide centric result") are the primary outputs that should have a checkmark ? The other outputs will still be visible in the history as long as the eye icon is shown (and that is fine).

**Comment by bernt-matthias** [comment]
Thanks for the feedback. I unchecked all, but the outputs of the MetaProSIP tool.

**Comment by bernt-matthias** [comment]
@timosachsenberg: Where to proceed once this one is finished? Would it be an option for you to open an issue here listing the workflows that are already available (e.g. for knime) that we could work on? Maybe linking the current state and other info. 

You mentioned once that many of them have Rscipt steps in them. Is this correct?

**Pull request merged**